### PR TITLE
Cleanup, more improvements to error messages.

### PR DESCRIPTION
**Pull request by prideout**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -31,8 +31,6 @@ function print_help {
     echo "        Platform(s) to build, defaults to desktop."
     echo "        Building for Android or iOS will automatically generate / download"
     echo "        the toolchains if needed and perform a partial desktop build."
-    echo "    -r"
```
Comment (romainguy) [inline build.sh]: Did we remove the feature? I thought we had issues in CI (macOS I believe with too many open files).

Comment (prideout) [reply build.sh]: If we ever supported this feature, it was before our GitHub history began.  Yeah I vaguely remember that issue.  Hm.

Comment (romainguy) [reply build.sh]: Is it gone from the rest of the script? We definitely used it in the Kokoro build scripts.

Comment (prideout) [reply build.sh]: Correct.

**Pull request merged**